libexec/rtld-elf/aarch64/rtld_c18n_machdep.c

```diff
@@ -178,16 +168,19 @@ tramp_compile(char **entry, const struct tramp_data *data)
 		}
 		COPY(invoke_res);
 	}
-	PATCH_ADR(cookie_off, size);
+	/*
+	 * Add 1 to offset due to capmode.
+	 */
+	PATCH_ADR(cookie_off, size + 1);
```

Comment:
This deserves a better comment to explain why the + 1 is there even for capmode (and please call it C64; capmode is the RISC-V name). If I understand correctly, it's because you use `LR == cookie` to detect a tail call, and since the benchmark ABI only clears the LSB of LR when it returns you need the cookie to match that, even though that's not the actual address it would use for a return.

There's probably a better name for this than `cookie`, too, that conveys what it's for. Otherwise it sounds like it's just an arbitrary bit of identifying data.